app/models/site.rb

```diff
@@ -62,12 +62,17 @@ def lon=(val)
     end
   end
 
+  # Returns true if the site geometry is null or has type 'ST_Point'.
+  def point?
+    geometry.nil? || geometry.geometry_type.type_name == 'Point'
+  end
+
```

Comment:
It is primarily a way to ensure we don't destroy geographical data from non-point sites by replacing them with their centroid when they are edited via Rails, which is what happens now.  We don't have to worry about destroying data if the geometry is already a point or if it is null.  I guess you could argue that "point?" is misnamed since if it's NULL it's not really a point, but this is no more offensive it seems than SQL's generally cavalier semantics surrounding the use of NULL.

The Rails code currently prohibits saving a site that doesn't have geographical data, but this hasn't always been the case (otherwise this kludge would be unnecessary—this is one reason I've been pushing hard for sensible database constraints over the past year or so).  26 out of the 1049 sites in the ebi_production database have NULL in the geometry column.